src/main/java/com/vestmark/nexus/plugin/maven/MavenResource.java

```diff
@@ -184,6 +187,19 @@ public Response download(
       response.header(HttpHeaders.CONTENT_DISPOSITION, String.format("attachment;filename=\"%s\"", fileName));
       return response.build();
     }
+    catch (RuntimeException e) {
+      log.error("storageTx response exception: ", e);
+      log.error(
+          "query params: {} {} {}-{}{}.{}",
+          repositoryName,
+          groupId,
+          artifactId,
+          version,
+          StringUtils.isBlank(classifier) ? "" : "-" + classifier,
+          extension);
+      storageTx.rollback();
+      throw e;
+    }
     finally {
       if (storageTx != null) {
         if (storageTx.isActive()) {
```

Comment:
If an exception is thrown on the `commit()` call, the `close()` call would be avoided.  If there was an exception thrown, would that show up in the logs?